tests/lib/parse.js

```diff
@@ -78,6 +78,93 @@ describe("parse()", () => {
+        });
+
+        it("should output the same value for program.start, end and loc when there is a leading comments with loc", () => {
+            const ast = espree.parse("/* foo */ bar", {
+                loc: true
+            });
+
+            assert.strictEqual(ast.start, ast.loc.start.column);
+            assert.strictEqual(ast.end, ast.loc.end.column);
+        });
+
+        it("should output the same value for program.start, end and range when there is a trailing comments", () => {
+            const ast = espree.parse(" bar /* foo */", {
+                range: true
+            });
+
+            assert.strictEqual(ast.start, ast.range[0]);
+            assert.strictEqual(ast.end, ast.range[1]);
+        });
+
+        it("should output the same value for range and start and end in templateElement with range", () => {
+            const ast = espree.parse("`foo ${bar}`;", {
+                ecmaVersion: 6,
+                range: true
+            });
+
+            assert.strictEqual(ast.body[0].expression.quasis[0].start, ast.body[0].expression.quasis[0].range[0]);
+            assert.strictEqual(ast.body[0].expression.quasis[0].end, ast.body[0].expression.quasis[0].range[1]);
+            assert.strictEqual(ast.body[0].expression.quasis[1].start, ast.body[0].expression.quasis[1].range[0]);
+            assert.strictEqual(ast.body[0].expression.quasis[1].end, ast.body[0].expression.quasis[1].range[1]);
+        });
+
+        it("should output the same value for loc and start and end in templateElement with loc", () => {
+            const ast = espree.parse("`foo ${bar}`;", {
+                ecmaVersion: 6,
+                loc: true
+            });
+
+            assert.strictEqual(ast.body[0].expression.quasis[0].start, ast.body[0].expression.quasis[0].loc.start.column);
+            assert.strictEqual(ast.body[0].expression.quasis[0].end, ast.body[0].expression.quasis[0].loc.end.column);
+            assert.strictEqual(ast.body[0].expression.quasis[1].start, ast.body[0].expression.quasis[1].loc.start.column);
+            assert.strictEqual(ast.body[0].expression.quasis[1].end, ast.body[0].expression.quasis[1].loc.end.column);
+        });
+
+        it("should output the same value for loc, range and start and end in templateElement with loc and range", () => {
+            const ast = espree.parse("`foo ${bar}`;", {
+                ecmaVersion: 6,
+                loc: true,
+                range: true
+            });
+
+            assert.strictEqual(ast.body[0].expression.quasis[0].start, ast.body[0].expression.quasis[0].loc.start.column);
+            assert.strictEqual(ast.body[0].expression.quasis[0].end, ast.body[0].expression.quasis[0].loc.end.column);
+            assert.strictEqual(ast.body[0].expression.quasis[1].start, ast.body[0].expression.quasis[1].range[0]);
+            assert.strictEqual(ast.body[0].expression.quasis[1].end, ast.body[0].expression.quasis[1].range[1]);
+        });
```

Comment:
This test is a bit confusing, `quasis[0]` start/end is compared to `loc` only, while `quasis[1]` to `range` only.